service/history/task/timer_active_task_executor.go

```diff
@@ -87,7 +87,7 @@ func (t *timerActiveTaskExecutor) Execute(
 		return nil
 	}
 
-	ctx, cancel := context.WithTimeout(context.Background(), taskDefaultTimeout)
+	ctx, cancel := context.WithTimeout(t.ctx, taskDefaultTimeout)
```

Comment:
this line is unnecessary for `TaskTypeDeleteHistoryEvent` case. so maybe create the context within each case below with corresponding timeout value.